### PR TITLE
chore: add another Chainsaw blog

### PR DESCRIPTION
**Pull request by eddycharly**
Add another Chainsaw blog.

### CODE REVIEW

**Review by chipzoller — COMMENTED** [review]
I think it would be best to space these out a bit more. Wait until next week.

**Comment by eddycharly** [comment]
> I think it would be best to space these out a bit more. Wait until next week.

Yup, would be nice to get your review though. I definitely want to put a spotlight on what assertion trees can do.

I'm afraid it's not understood well yet, IMHO users didn't realise that assertions are more than just YAML.

File: content/en/blog/general/why-chainsaw-is-unique/index.md [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+    k8s-app: kube-dns
+  namespace: kube-system
+spec:
+  template:
+    spec:
+      (containers[?securityContext == null]):
+        (len(@)): 0
+```
+
+In the assertion above, the first three fileds `spec`, `template`, and `spec` are basic projections that simply take the content of their respective fields and pass that to descendants.
```
Comment (vishal-chdhry) [inline content/en/blog/general/why-chainsaw-is-unique/index.md:126]:
```suggestion
In the assertion above, the first three fields `spec`, `template`, and `spec` are basic projections that simply take the content of their respective fields and pass that to descendants.
```

File: content/en/blog/general/why-chainsaw-is-unique/index.md [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+
+`(containers[?securityContext == null])` is a JMESPath expression filtering the `containers` array, selecting only the element where `securityContext` is `null`. This projection results in a new array that is passed to the descendant (`(len(@))` in this case).
+
+Finally `(len(@))` is reached, it's another JMESPath expression that computes the lenght of the array. There's no more descendant at this point, we're of a leaf of the YAML tree, the array length we just computed is then compared against 0.
```
Comment (vishal-chdhry) [inline content/en/blog/general/why-chainsaw-is-unique/index.md:130]:
```suggestion
Finally `(len(@))` is reached, it's another JMESPath expression that computes the length of the array. There's no more descendant at this point, we're of a leaf of the YAML tree, the array length we just computed is then compared against 0.
```

File: content/en/blog/general/why-chainsaw-is-unique/index.md [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+  template:
+    spec:
+      # the ~ modifier tells Chainsaw to iterate over the array elements
+      ~.(containers):
+        securityContext: {}
+```
+
+This assertion uses the `~` modifier and Chainsaw will evaluate descendants once per element in the array.
+
+### Explict bindings and reference
```
Comment (vishal-chdhry) [inline content/en/blog/general/why-chainsaw-is-unique/index.md:161]:
```suggestion
### Explicit bindings and reference
```

File: content/en/blog/general/why-chainsaw-is-unique/index.md [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+
+In the assertion above, `(spec)->specBinding` means that Chainsaw will keep a reference to the result of the expression evaluation and will make it available to descendants under the `$specBinding` binding.
+
+Now we can use the binding to write the assertion check taking the pod level security context into account like this `($specBinding.securityContext != null || securityContext != null): true`.
+
+### Give me more
+
+We covered all major features of assertion trees in this blog post.
+
+Chainsaw doesn't directly implements assertion trees but relies on the [kyverno-json package](https://kyverno.github.io/kyverno-json/policies/asserts/#assertion-trees). You can browse this documentation to learn more about assertion trees, this documentation also applies to Chainsaw.
```
Comment (vishal-chdhry) [inline content/en/blog/general/why-chainsaw-is-unique/index.md:190]:
```suggestion
Chainsaw doesn't directly implement assertion trees but relies on the [kyverno-json package](https://kyverno.github.io/kyverno-json/policies/asserts/#assertion-trees). You can browse this documentation to learn more about assertion trees, this documentation also applies to Chainsaw.
```

File: content/en/blog/general/why-chainsaw-is-unique/index.md [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+
+JMESpath supports functions and also allows custom functions to be registered in the JMESPath interpreter. The supported functions list is available in this [documentation page](https://kyverno.github.io/chainsaw/latest/jp/functions/).
+
+## Conclusion
+
+I hope this blog post will help understanding what assertion trees are and how they work.
```
Comment (vishal-chdhry) [inline content/en/blog/general/why-chainsaw-is-unique/index.md:196]:
```suggestion
I hope this blog post will help understand what assertion trees are and how they work.
```

File: content/en/blog/general/why-chainsaw-is-unique/index.md [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+  labels:
+    k8s-app: kube-dns
+  namespace: kube-system
+spec:
+  replicas: 2
+```
+
+This time we are not providing a resource name.
+
+Chainsaw will lookup all Deployments with the `k8s-app: kube-dns` label in the `kube-system` namespace. The assertion will be considered valid if **at least** one Deployment matches with the (partial) resource definition contained in the assertion, if none match the assertion will be considered failed.
```
Comment (vishal-chdhry) [inline content/en/blog/general/why-chainsaw-is-unique/index.md:64]:
```suggestion
Chainsaw will lookup all Deployments with the `k8s-app: kube-dns` label in the `kube-system` namespace. The assertion will be considered valid if **at least** one Deployment matches with the (partial) resource definition contained in the assertion, if none matches the assertion will be considered failed.
```

File: content/en/blog/general/why-chainsaw-is-unique/index.md [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+
+Apart from the resource lookup process being a little bit more interesting, this kind of assertion is essentially the same as the previous one. It's basically taking a decision by comparing an actual and expected resource.
+
+### A word on timeout
+
+Keep in mind that Chainsaw won't consider an assertion valid or failed in one shot. The assertion will constinuously evaluated during a predetermined period until the assertion succeeds or the timeout expires.
```
Comment (vishal-chdhry) [inline content/en/blog/general/why-chainsaw-is-unique/index.md:70]:
```suggestion
Keep in mind that Chainsaw won't consider an assertion valid or failed in one shot. The assertion will continuously evaluated during a predetermined period until the assertion succeeds or the timeout expires.
```

**Review by vishal-chdhry — COMMENTED** [review]
Some minor typos

**Comment by eddycharly** [comment]
> There is a [grammarly VSCode](https://marketplace.visualstudio.com/items?itemName=znck.grammarly) extension

Just installed it, everything is red now 😂

File: content/en/blog/general/why-chainsaw-is-unique/index.md [hunk 8]
```diff
@@ -0,0 +1,200 @@
...
+
+In the assertion above, `(spec)->specBinding` means that Chainsaw will keep a reference to the result of the expression evaluation and will make it available to descendants under the `$specBinding` binding.
+
+Now we can use the binding to write the assertion check taking the pod level security context into account like this `($specBinding.securityContext != null || securityContext != null): true`.
+
+### Give me more
+
+We covered all major features of assertion trees in this blog post.
+
+Chainsaw doesn't directly implement assertion trees but relies on the [kyverno-json package](https://github.com/kyverno/kyverno-json). You can browse [this documentation](https://kyverno.github.io/kyverno-json/policies/asserts/#assertion-trees) to learn more about assertion trees, this also applies to Chainsaw.
```
Comment (realshuting) [inline content/en/blog/general/why-chainsaw-is-unique/index.md:190]: It would be super helpful to link to the playground as well https://kyverno.github.io/kyverno-json/_playground/.

File: content/en/blog/general/why-chainsaw-is-unique/index.md [hunk 9]
```diff
@@ -0,0 +1,200 @@
...
+metadata:
+  name: coredns
+  namespace: kube-system
+spec:
+  replicas: 2
+```
+
+When asking Chainsaw to execute the assertion above, it will look for a Deployment named `coredns` in the `kube-system` namespace and will compare the existing resource with the (partial) resource definition contained in the assertion.
+
+In this specific case, if `spec.replicas` is set to 2 in the existing resource, the assertion will be considered valid, if it is not equal to 2 the assertion will be considered failed.
```
Comment (chipzoller) [inline content/en/blog/general/why-chainsaw-is-unique/index.md:43]:
```suggestion
In this specific case, if `spec.replicas` is set to 2 in the existing resource, the assertion will be considered valid. If it is not equal to 2 the assertion will be considered failed.
```

File: content/en/blog/general/why-chainsaw-is-unique/index.md [hunk 10]
```diff
@@ -0,0 +1,200 @@
...
+
+Apart from the resource lookup process being a little bit more interesting, this kind of assertion is essentially the same as the previous one. Chainsaw is basically taking a decision by comparing an actual and expected resource.
+
+### A word on timeout
+
+Keep in mind that Chainsaw won't consider an assertion valid or failed in one shot. The assertion will be continuously evaluated during a predetermined period of time until the assertion succeeds or the timeout expires.
+
+## It's good but it's not enough
+
+Now we covered the basic assertions we can look at their limitations and how Chainsaw solves them.
```
Comment (chipzoller) [inline content/en/blog/general/why-chainsaw-is-unique/index.md:74]:
```suggestion
Now that we covered the basic assertions we can look at their limitations and how Chainsaw solves them.
```

File: content/en/blog/general/why-chainsaw-is-unique/index.md [hunk 11]
```diff
@@ -0,0 +1,200 @@
...
+Basic assertions cover the simple cases but:
+
+- Working with arrays is not easy
+    - How are we supposed to compare arrays with different sizes?
+    - How can we check a specific item in an array is present?
+- Matching exact values is not always what we want
+    - How can we verify the number of replicas is above a certain number (not exactly this number)?
+    - How can we apply a regex to a label?
+
+The examples above are the most obvious ones, still they demonstrate that an assertion model needs to be more rich and more flexible than simple comparisons.
```
Comment (chipzoller) [inline content/en/blog/general/why-chainsaw-is-unique/index.md:85]:
```suggestion
While the examples above are the most obvious ones, they demonstrate that an assertion model needs to be more rich and more flexible than simple comparisons.
```

File: content/en/blog/general/why-chainsaw-is-unique/index.md [hunk 12]
```diff
@@ -0,0 +1,200 @@
...
+
+### Overview of the Chainsaw Assertion Model
+
+Let's see how Chainsaw can assert that the number of replicas of a Deployment is greater than a certain threshold, let's look at how it can be written.
```
Comment (chipzoller) [inline content/en/blog/general/why-chainsaw-is-unique/index.md:89]:
```suggestion
Let's see how Chainsaw can assert that the number of replicas of a Deployment is greater than a certain threshold. Let's look at how it can be written.
```

run-on

File: content/en/blog/general/why-chainsaw-is-unique/index.md [hunk 13]
```diff
@@ -0,0 +1,200 @@
...
+kind: Deployment
+metadata:
+  labels:
+    k8s-app: kube-dns
+  namespace: kube-system
+spec:
+  (replicas > 2): true
+```
+
+In the assertion above the **key part** of the YAML syntax contains a [JMESPath](https://jmespath.site) expression, the evaluation of this expression will be compared to the **value part** of the YAML syntax.
```
Comment (chipzoller) [inline content/en/blog/general/why-chainsaw-is-unique/index.md:102]:
```suggestion
In the assertion above, the **key part** of the YAML syntax contains a [JMESPath](https://jmespath.site) expression. The evaluation of this expression will be compared to the **value part** of the YAML syntax.
```

run-on

File: content/en/blog/general/why-chainsaw-is-unique/index.md [hunk 14]
```diff
@@ -0,0 +1,200 @@
...
+
+In this case, the expression is simple but represents the main concept behind the Chainsaw assertion model. At every node in the YAML tree, Chainsaw can create a projection of the being processed element and pass the result of this projection to the descendants. When a leaf of the tree is reached a comparison happens and if the comparison fails the assertion will be considered failed. If all comparisons succeed the assertion will be considered valid.
```
Comment (chipzoller) [inline content/en/blog/general/why-chainsaw-is-unique/index.md:104]:
```suggestion
In this case, the expression is simple but represents the main concept behind the Chainsaw assertion model. At every node in the YAML tree, Chainsaw can create a projection of the element currently being processed and pass the result of this projection to the descendants. When a leaf of the tree is reached, a comparison happens. If the comparison fails, the assertion will be considered failed. If all comparisons succeed, the assertion will be considered valid.
```

File: content/en/blog/general/why-chainsaw-is-unique/index.md [hunk 15]
```diff
@@ -0,0 +1,200 @@
...
+    spec:
+      (containers[?securityContext == null]):
+        (len(@)): 0
+```
+
+In the assertion above, the first three fields `spec`, `template`, and `spec` are basic projections that simply take the content of their respective fields and pass that to descendants.
+
+`(containers[?securityContext == null])` is a JMESPath expression filtering the `containers` array, selecting only the element where `securityContext` is `null`. This projection results in a new array that is passed to the descendant (`(len(@))` in this case).
+
+Finally `(len(@))` is reached, it's another JMESPath expression that computes the length of the array. There's no more descendant at this point, we're of a leaf of the YAML tree, and the array length we just computed is then compared against 0.
```
Comment (chipzoller) [inline content/en/blog/general/why-chainsaw-is-unique/index.md:130]:
```suggestion
`(len(@))` is another JMESPath expression that computes the length of the array. There's no more descendant at this point. We're at a leaf of the YAML tree and the array length we just computed is then compared against 0.
```

File: content/en/blog/general/why-chainsaw-is-unique/index.md [hunk 16]
```diff
@@ -0,0 +1,200 @@
...
+
+If the comparison matches, the assertion will be considered valid, if not it will be considered failed.
```
Comment (chipzoller) [inline content/en/blog/general/why-chainsaw-is-unique/index.md:132]:
```suggestion
If the comparison matches, the assertion will be considered valid; if not, it will be considered failed.
```

File: content/en/blog/general/why-chainsaw-is-unique/index.md [hunk 17]
```diff
@@ -0,0 +1,200 @@
...
+
+In simple words, the assertion above checks that all containers in the Deployment have a security context configured.
+
+### Iterating over arrays (or maps)
+
+Being able to filter arrays allows selecting the elements to be processed.
+
+On top of that Chainsaw allows iterating over array elements to validate each item separately.
```
Comment (chipzoller) [inline content/en/blog/general/why-chainsaw-is-unique/index.md:140]:
```suggestion
On top of that, Chainsaw allows iterating over array elements to validate each item separately.
```

File: content/en/blog/general/why-chainsaw-is-unique/index.md [hunk 18]
```diff
@@ -0,0 +1,200 @@
...
+        securityContext: {}
+```
+
+This assertion uses the `~` modifier and Chainsaw will evaluate descendants once per element in the array.
+
+### Explicit bindings and reference
+
+When passing from a parent node to descendant nodes, it can be useful to keep a reference to one or more parent nodes.
+
+Let's say we want to consider the pod level security context too, we can write such an assertion with the syntax below:
```
Comment (chipzoller) [inline content/en/blog/general/why-chainsaw-is-unique/index.md:165]:
```suggestion
Let's say we want to consider the pod-level security context in addition to the container-level security context. We can write such an assertion with the syntax below:
```

run-on

File: content/en/blog/general/why-chainsaw-is-unique/index.md [hunk 19]
```diff
@@ -0,0 +1,200 @@
...
+  template:
+    (spec)->specBinding:
+      # the ~ modifier tells Chainsaw to iterate over the array elements
+      ~.(containers):
+        ($specBinding.securityContext != null || securityContext != null): true
+```
+
+In the assertion above, `(spec)->specBinding` means that Chainsaw will keep a reference to the result of the expression evaluation and will make it available to descendants under the `$specBinding` binding.
+
+Now we can use the binding to write the assertion check taking the pod level security context into account like this `($specBinding.securityContext != null || securityContext != null): true`.
```
Comment (chipzoller) [inline content/en/blog/general/why-chainsaw-is-unique/index.md:184]: This is complex. You should write at least one sentence describing, in plain English, what this assertion does. Based on studying the example and what you wrote, it's something like: "This assertion checks to ensure that, if any container's securityContext is not defined, the Pod-level securityContext is defined."

File: content/en/blog/general/why-chainsaw-is-unique/index.md (hunk 8 above)
Comment (chipzoller) [inline content/en/blog/general/why-chainsaw-is-unique/index.md:190]:
```suggestion
Chainsaw doesn't directly implement assertion trees but relies on the [kyverno-json package](https://github.com/kyverno/kyverno-json). You can browse [the documentation](https://kyverno.github.io/kyverno-json/policies/asserts/#assertion-trees) to learn more about assertion trees which also applies to Chainsaw.
```

File: content/en/blog/general/why-chainsaw-is-unique/index.md [hunk 20]
```diff
@@ -0,0 +1,200 @@
...
+
+### Give me more
+
+We covered all major features of assertion trees in this blog post.
+
+Chainsaw doesn't directly implement assertion trees but relies on the [kyverno-json package](https://github.com/kyverno/kyverno-json). You can browse [this documentation](https://kyverno.github.io/kyverno-json/policies/asserts/#assertion-trees) to learn more about assertion trees, this also applies to Chainsaw.
+
+JMESpath supports functions and also allows custom functions to be registered in the JMESPath interpreter. The supported functions list is available on this [documentation page](https://kyverno.github.io/chainsaw/latest/jp/functions/).
```
Comment (chipzoller) [inline content/en/blog/general/why-chainsaw-is-unique/index.md:192]:
```suggestion
JMESPath supports functions and also allows custom functions to be registered in the JMESPath interpreter. The supported list of functions is available [here](https://kyverno.github.io/chainsaw/latest/jp/functions/).
```

File: content/en/blog/general/why-chainsaw-is-unique/index.md [hunk 21]
```diff
@@ -0,0 +1,200 @@
...
+
+## Conclusion
+
+I hope this blog post will help understand what assertion trees are and how they work.
```
Comment (chipzoller) [inline content/en/blog/general/why-chainsaw-is-unique/index.md:196]:
```suggestion
I hope this blog post helps you understand what assertion trees are and how they work.
```

File: content/en/blog/general/why-chainsaw-is-unique/index.md [hunk 22]
```diff
@@ -0,0 +1,200 @@
...
+
+Simple tests usually don't need that level of flexibility but it comes very handy when your tests need to become more complex.
```
Comment (chipzoller) [inline content/en/blog/general/why-chainsaw-is-unique/index.md:198]:
```suggestion
Simple tests usually only need simple capabilities, but assertion trees can come in very handy when your tests become more complex.
```

**Comment by eddycharly** [comment]
Thanks for the review folks.

File: content/en/blog/general/why-chainsaw-is-unique/index.md [hunk 23]
```diff
@@ -0,0 +1,200 @@
...
+
+- This is hard to make it shine out of pure documentation
+- You can't appreciate Chainsaw until you understand what makes it so unique
+- Seeing the feature in action is the best way to learn about it
+
+What makes Chainsaw unique is its **_assertion model_**.
+
+In this blog post we will deep dive into the assertion model used by Chainsaw, how it works, and how we extended something that looks like YAML but is more than that.
+
+After all, if Chainsaw was only about applying resources in a cluster and checking fields are set correctly on those resources it would be deceptive, some tools doing just that already exist and Chainsaw would be more or less a duplicate of those tools.
```
[the remark this answers is not in the document]
Comment (eddycharly) [reply content/en/blog/general/why-chainsaw-is-unique/index.md:21]: This is a run-on sentence right ? Hence the `.` punctuation you added ?

[the remark this answers is not in the document]
Comment (chipzoller) [reply content/en/blog/general/why-chainsaw-is-unique/index.md:21]: Yes, those are two independent sentences.

File: content/en/blog/general/why-chainsaw-is-unique/index.md [hunk 24]
```diff
@@ -0,0 +1,200 @@
...
+  namespace: kube-system
+spec:
+  replicas: 2
+```
+
+This time we are not providing a resource name.
+
+Chainsaw will look up all Deployments with the `k8s-app: kube-dns` label in the `kube-system` namespace. The assertion will be considered valid if **at least one** Deployment matches with the (partial) resource definition contained in the assertion, if none matches the assertion will be considered failed.
+
+Apart from the resource lookup process being a little bit more interesting, this kind of assertion is essentially the same as the previous one. Chainsaw is basically taking a decision by comparing an actual and expected resource.
```
[the remark this answers is not in the document]
Comment (eddycharly) [reply content/en/blog/general/why-chainsaw-is-unique/index.md:66]: I was not sure, thanks.

File: content/en/blog/general/why-chainsaw-is-unique/index.md (hunk 12 above)
Comment (eddycharly) [reply content/en/blog/general/why-chainsaw-is-unique/index.md:89]: `Let's` appears twice in this short sentence.

File: content/en/blog/general/why-chainsaw-is-unique/index.md (hunk 23 above)
[the remark this answers is not in the document]
Comment (eddycharly) [reply content/en/blog/general/why-chainsaw-is-unique/index.md:21]: 👍

File: content/en/blog/general/why-chainsaw-is-unique/index.md [hunk 25]
```diff
@@ -0,0 +1,200 @@
...
+      # the ~ modifier tells Chainsaw to iterate over the array elements
+      ~.(containers):
+        ($specBinding.securityContext != null || securityContext != null): true
+```
+
+In the assertion above, `(spec)->specBinding` means that Chainsaw will keep a reference to the result of the expression evaluation and will make it available to descendants under the `$specBinding` binding.
+
+Now we can use the binding to write the assertion check taking the pod level security context into account like this `($specBinding.securityContext != null || securityContext != null): true`.
+
+### Give me more
```
Comment (eddycharly) [inline content/en/blog/general/why-chainsaw-is-unique/index.md:186]: I don't like this title, @chipzoller do you have a suggestion for a better one ?

Comment (eddycharly) [reply content/en/blog/general/why-chainsaw-is-unique/index.md:186]: (above is a run-on sentence ? ;-) )

Comment (eddycharly) [reply content/en/blog/general/why-chainsaw-is-unique/index.md:186]: I don't like this title. @chipzoller do you have a suggestion for a better one ?

File: content/en/blog/general/why-chainsaw-is-unique/index.md [hunk 26]
```diff
@@ -0,0 +1,200 @@
+---
+date: 2023-12-13
+title: Kyverno Chainsaw - The one and only thing making Chainsaw unique!
+linkTitle: Kyverno Chainsaw - The one and only thing making Chainsaw unique
```
[the remark this answers is not in the document]
Comment (eddycharly) [reply content/en/blog/general/why-chainsaw-is-unique/index.md:4]: Let me ask ChatGPT 😂

[the remark this answers is not in the document]
Comment (eddycharly) [reply content/en/blog/general/why-chainsaw-is-unique/index.md:4]: Kyverno Chainsaw - Exploring the Unique Assertion Model that Sets it Apart

File: content/en/blog/general/why-chainsaw-is-unique/index.md (hunk 8 above)
Comment (eddycharly) [reply content/en/blog/general/why-chainsaw-is-unique/index.md]: Thanks, done.

File: content/en/blog/general/why-chainsaw-is-unique/index.md (hunk 19 above)
Comment (eddycharly) [reply content/en/blog/general/why-chainsaw-is-unique/index.md]: Done, thanks.

File: content/en/blog/general/why-chainsaw-is-unique/index.md (hunk 26 above)
[the remark this answers is not in the document]
Comment (eddycharly) [reply content/en/blog/general/why-chainsaw-is-unique/index.md:4]: Kyverno Chainsaw - Exploring the Assertion Model that Makes it Unique

[the remark this answers is not in the document]
Comment (chipzoller) [reply content/en/blog/general/why-chainsaw-is-unique/index.md]: Kyverno Chainsaw: Assertion Models

File: content/en/blog/general/why-chainsaw-is-unique/index.md (hunk 25 above)
Comment (eddycharly) [reply content/en/blog/general/why-chainsaw-is-unique/index.md:187]: ## What else

Comment (chipzoller) [reply content/en/blog/general/why-chainsaw-is-unique/index.md:187]: "More" is fine.

Comment (eddycharly) [reply content/en/blog/general/why-chainsaw-is-unique/index.md]: Like What more ?

File: content/en/blog/general/why-chainsaw-is-unique/index.md [hunk 27]
```diff
@@ -0,0 +1,202 @@
...
+
+Chainsaw will look up all Deployments with the `k8s-app: kube-dns` label in the `kube-system` namespace. The assertion will be considered valid if **at least one** Deployment matches with the (partial) resource definition contained in the assertion. If none matches the assertion will be considered failed.
+
+Apart from the resource lookup process being a little bit more interesting, this kind of assertion is essentially the same as the previous one. Chainsaw is basically making a decision by comparing an actual and expected resource.
+
+### A word on timeout
+
+Keep in mind that Chainsaw won't consider an assertion valid or failed in one shot. The assertion will be continuously evaluated during a predetermined period of time until the assertion succeeds or the timeout expires.
+
+## It's good but it's not enough
```
Comment (JimBugwadia) [inline content/en/blog/general/why-chainsaw-is-unique/index.md:72]:
```suggestion
## It's good, but not enough
```

File: content/en/blog/general/why-chainsaw-is-unique/index.md [hunk 28]
```diff
@@ -0,0 +1,202 @@
...
+
+- This is hard to make it shine out of pure documentation
+- You can't appreciate Chainsaw until you understand what makes it so unique
+- Seeing the feature in action is the best way to learn about it
+
+What makes Chainsaw unique is its **_assertion model_**.
+
+In this blog post we will deep dive into the assertion model used by Chainsaw, how it works, and how we extended something that looks like YAML but is more than that.
+
+After all, if Chainsaw was only about applying resources in a cluster and checking fields are set correctly on those resources it would be deceptive. There are tools which can do that today and Chainsaw would be more or less redundant.
```
[the remark this answers is not in the document]
Comment (eddycharly) [reply content/en/blog/general/why-chainsaw-is-unique/index.md]: @JimBugwadia do you suggest to remove this paragraph ?
